pkg/controllers/state/cluster.go

```diff
@@ -339,11 +339,11 @@ func (c *Cluster) updateNodeUsageFromPod(pod *v1.Pod) {
 }
 
 func (c *Cluster) getInstanceType(ctx context.Context, provisioner *v1alpha5.Provisioner, instanceTypeName string) (cloudprovider.InstanceType, error) {
-	if provisioner == nil || provisioner.Spec.Provider == nil {
+	if provisioner == nil || (provisioner.Spec.Provider == nil && provisioner.Spec.ProviderRef == nil) {
```

Comment:
Is this necessary given our validation? We can assume one of these is populated as an invariant of the codebase. 
```suggestion
	if provisioner == nil {
```

Reply:
Yeah makes sense. With the earlier check, what was happening was that we'd return early since Provider was indeed nil (since ProviderRef was being used instead) and that was causing in-flight support to break.